src/services/ruleEngine/DateCalculator.ts

```
@@ -0,0 +1,413 @@
+    const now = new Date();
+    const pattern = config.calculation; // e.g., "monthly", "quarterly", "yearly"
+    
+    let nextDate: Date;
+    
+    switch (pattern) {
+      case 'monthly':
+        nextDate = new Date(now.getFullYear(), now.getMonth() + 1, 1);
+        break;
+        
+      case 'quarterly':
+        const currentQuarter = Math.floor(now.getMonth() / 3);
```

Biome (2.1.2)

[error] Other switch clauses can erroneously access this declaration.
Wrap the declaration in a block to restrict its access to the switch clause.

The declaration is defined in this switch clause:

Safe fix: Wrap the declaration in a block.


(lint/correctness/noSwitchDeclarations)